File: packages/daimo-expo-enclave/android/src/main/java/expo/modules/enclave/KeyManager.kt
package expo.modules.enclave

import expo.modules.core.Promise
import expo.modules.core.arguments.ReadableArguments

interface KeyManager {
  fun createKeyPair(accountName: String): String
  fun deleteKeyPair(accountName: String)
  fun fetchPublicKey(accountName: String): String?
  fun sign(accountName: String, hexMessage: String, promptCopy: ReadableArguments, promise: Promise)
  fun verify(accountName: String, hexSignature: String, hexMessage: String): Boolean
}

File: packages/daimo-expo-enclave/ios/KeyManager.swift
//
//  KeyManager.swift
//  Daimo
//
//  Created by Nalin Bhardwaj.
//  Copyright © 2023 Daimo. All rights reserved.
//
//  The common protocol for key management (creation, sign, verify, fetch etc.).

protocol KeyManager {
    func fetchPublicKey(accountName: String) throws -> String?
    func createKeyPair(accountName: String) throws -> String
    func deleteKeyPair(accountName: String) throws
    func sign(accountName: String, hexMessage: String, usageMessage: String) throws -> String
    func verify(accountName: String, hexSignature: String, hexMessage: String) throws -> Bool
}